sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java

```diff
@@ -140,6 +149,66 @@ public void testNullSubscription() {
     assertNotNull(DisplayData.from(read));
   }
 
+  /**
+   * {@link PipelineOptions} with a ValueProvider.
+   */
+  public interface RuntimeSubscriptionOptions extends PipelineOptions {
```

Comment:
Can you use TestPipeline.newProvider() instead?

Reply:
Yes. Done.